src/windows.rs

```diff
@@ -1,29 +1,69 @@
 use std::{
     ffi::{OsStr, OsString},
-    process::Command,
+    fmt::Debug,
+    io,
+    process::{Command, Stdio},
 };
 
 use std::os::windows::process::CommandExt;
 
 const CREATE_NO_WINDOW: u32 = 0x08000000;
 
-pub fn commands<T: AsRef<OsStr>>(path: T) -> Vec<Command> {
-    let mut cmd = Command::new("cmd");
+pub enum WindowsRunCommand {
+    Start(Command),
+    Explorer(Command),
+}
+
+impl WindowsRunCommand {
+    pub fn status_without_output(&mut self) -> io::Result<std::process::ExitStatus> {
+        let cmd = match self {
+            WindowsRunCommand::Start(cmd) => cmd,
+            WindowsRunCommand::Explorer(cmd) => cmd,
+        };
+        cmd.stdin(Stdio::null())
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .status()
+    }
+
+    pub fn extract(self) -> Command {
+        match self {
+            WindowsRunCommand::Start(cmd) => cmd,
+            WindowsRunCommand::Explorer(cmd) => cmd,
+        }
+    }
+}
+
+impl Debug for WindowsRunCommand {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Start(_) => write!(f, "start"),
```

Comment:
I think it would be better to delegate to the inner `cmd` here, as every implementation prints everything it tries to execute. Windows shouldn't be different.